src/components/Table/Table.jsx

```diff
@@ -42,21 +56,31 @@ const TableComponent = (props) => {
         </TableHead>
         <TableBody>
           {data.map((trainee) => (
-            <StyledTableRow key={trainee[id]} onClick={() => onSelect(trainee[id])}>
+            <StyledTableRow key={trainee[id]}>
               {
                 columns.map((column) => (
-                  <TableCell key={`${trainee[id]}${column.field}`} align={column.align}>
+                  <TableCell key={`${trainee[id]}${column.field}`} align={column.align} onClick={() => onSelect(trainee[id])}>
                     <Typography>
                       {column.format
                         ? column.format(trainee[column.field]) : trainee[column.field]}
                     </Typography>
                   </TableCell>
                 ))
               }
+              <TableCell align="center" key={`${trainee[id]}action`}>
+                {
+                  actions.map((action, index) => (
+                    <IconButton key={`action${index + 1}`} disableFocusRipple size="small" onClick={() => action.handler(trainee)}>
```

Comment:
why use action for making the key as action is an object just use index for the key or you can pass actions with the unique id's